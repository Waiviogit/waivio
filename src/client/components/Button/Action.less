@import (reference) '../../styles/custom.less';
@import (reference) '../../styles/mixins.less';

.Action {
  .btn;
  .btn-default;

  .colorize-button(#9faeba, #7c92a0);

  color: @white;

  &--primary {
    background: var(--website-color) !important;
    border-color: var(--website-color) !important;
    color: @white-darker;

    &:hover,
    &:focus,
    &:active,
    &.active {
      background: var(--website-color) !important;
      border-color: var(--website-color) !important;
      color: @white-darker !important;
    }
    &:disabled,
    &:disabled:hover {
      background-color: @white-smoke !important;
      border-color: @white-gainsboro !important;
      color: rgba(0, 0, 0, 0.25) !important;
      cursor: no-drop;
    }
  }

  &--big {
    padding: 0 15px;
    height: 46px;
  }

  &:hover,
  &:focus,
  &:active,
  &.active {
    color: @white;
  }

  & .anticon {
    vertical-align: -1px;
    margin-right: 8px;
  }
}
