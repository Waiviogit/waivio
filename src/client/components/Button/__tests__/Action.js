import React from 'react';
import { shallow, mount } from 'enzyme';
import Action from '../Action';

describe('<Action />', () => {
  it('renders without exploding', () => {
    const props = {
      loading: true,
      disabled: false,
      primary: true,
    };
    const wrapper = shallow(<Action {...props}>Example text</Action>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should include Action--primary in className', () => {
    const props = {
      loading: true,
      disabled: false,
      primary: true,
    };
    const wrapper = shallow(<Action {...props}>Example text</Action>);
    expect(wrapper.props().className).toBe('Action Action--primary');
  });

  it('should include Action--big in className', () => {
    const props = {
      loading: true,
      disabled: false,
      big: true,
    };
    const wrapper = shallow(<Action {...props}>Example text</Action>);
    expect(wrapper.props().className).toBe('Action Action--big');
  });

  it('should render Icon', () => {
    const props = {
      loading: true,
      disabled: false,
      big: true,
    };
    const wrapper = mount(<Action {...props}>Example text</Action>);
    expect(wrapper.find('Icon')).not.toBeUndefined();
  });

  it('should have Text as a child', () => {
    const props = {
      loading: true,
      disabled: false,
      big: true,
    };
    const wrapper = mount(<Action {...props}>Example text</Action>);
    expect(wrapper.text()).toBe('Example text');
  });
});
