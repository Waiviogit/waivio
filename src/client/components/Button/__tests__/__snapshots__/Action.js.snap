// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> renders without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Action
    big={false}
    className=""
    disabled={false}
    loading={true}
    onClick={[Function]}
    primary={true}
  >
    Example text
  </Action>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          type="loading"
        />,
        "Example text",
      ],
      "className": "Action Action--primary",
      "disabled": false,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": "loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Example text",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            type="loading"
          />,
          "Example text",
        ],
        "className": "Action Action--primary",
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "loading",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Example text",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
